.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseMergedDateTimeParser.cs

```diff
@@ -282,33 +282,33 @@ public DateTimeParseResult Parse(ExtractResult er, DateObject refTime)
                 var equalMatch = Config.EqualRegex.MatchBegin(er.Text, trim: true);
 
                 // check also after match
-                if (this.Config.CheckBothBeforeAfter)
+                if (this.Config.CheckBothBeforeAfter && er.Data != null && er.Data.Equals(Constants.HAS_MOD))

```

Comment:
I'm not sure if this change is not causing side-effects. We'll need to run more experiments.
Is there a specific reason the previous code was not correct for Turkish?
The semantics of the new has_mod and the previous type checks don't match.

Reply:
It turned out that the previous solution was specific to the Merged test cases and did not work for new test cases in DateTimeModel. The new approach is more general. 
The problem is that, in Turkish, these modifiers are usually found at the end of the match whether they are part of the already extracted subentity or added by MergedExtractor, but they need to be parsed separately in MergedParser only in the second case. 
Take, for example, "tomorrow before 10am" and "before tomorrow 10am", in Turkish, "before" would be at the end of the match in both cases.

Reply:
We're running some regression tests across languages to see if there's any side-effect. Thanks.